### PR TITLE
Add intent for getting current stop info

### PR DESCRIPTION
**Pull request by barbeau**
Knocking off one more item for the commands we want to support in v1.0 (#3).  I decided not to persist stop code, and just hit the OBA REST API for fresh stop code and stop name info, based on the saved `stopId`.

@philipmw Please take a look.

### CODE REVIEW

File: src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java [hunk 1]
```diff
@@ -90,6 +90,17 @@ public ObaUserClient(String obaBaseUrl) throws URISyntaxException {
     }
 
     /**
+     * Returns details about a particular stop, given it's stopId
+     * @param stopId
+     * @return details about a particular stop, given it's stopId
+     */
+    public ObaStopResponse getStopDetails(String stopId) {
+        ObaStopResponse response = new ObaStopRequest.Builder(stopId).build().call();
```
Comment (philipmw) [inline src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java]: Is it possible for this to be null?

Comment (barbeau) [inline src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java]: No, but that reminded me that we should be checking the HTTP code before assuming the result is valid.  Just pushed a new commit that includes this.

File: src/test/java/org/onebusaway/alexa/AuthedSpeechletTest.java [hunk 2]
```diff
@@ -59,6 +64,38 @@ public void initializeAuthedSpeechlet() throws URISyntaxException {
     }
 
     @Test
+    public void getStopDetails() throws SpeechletException, URISyntaxException {
```
Comment (barbeau) [inline src/test/java/org/onebusaway/alexa/AuthedSpeechletTest.java]: @philipmw I was trying to write a unit test for this, but I can't seem to get it to pass - it currently fails with `com.amazon.speech.speechlet.SpeechletException: Error getting stop details for 6497`.

What am I missing for the proper setup here?

Comment (philipmw) [inline src/test/java/org/onebusaway/alexa/AuthedSpeechletTest.java]: Cool, thanks for improving code coverage. :smile: 
I pushed a commit to your branch that fixes the unit test.

Comment (barbeau) [inline src/test/java/org/onebusaway/alexa/AuthedSpeechletTest.java]: Ah, thanks!  :)

**Comment by barbeau** [comment]
Ok, just squashed both commits to clean up this PR - should be ready to merge assuming Travis passes.

File: src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java [hunk 3]
```diff
@@ -90,6 +90,21 @@ public ObaUserClient(String obaBaseUrl) throws URISyntaxException {
...
 
     /**
+     * Returns details about a particular stop, given it's stopId
+     * @param stopId
+     * @return details about a particular stop, given it's stopId
+     */
+    public Optional<ObaStopResponse> getStopDetails(String stopId) {
+        ObaStopResponse response = new ObaStopRequest.Builder(stopId).build().call();
+        log.debug("ObaStopRequest returned " + response.toString());
+        if (response.getCode() == ObaApi.OBA_OK) {
```
Comment (philipmw) [inline src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java]: When might the response be anything but OBA_OK? If only when the OBA API is down, then I'd rather throw an exception here than force the client to unwrap the Optional.

Comment (barbeau) [inline src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java]: Ok - just force pushed a new commit that throws an exception if response isn't `OBA_OK`.  I'll update the other REST API calls with a similar format in a separate PR.

**Pull request merged**